src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -4468,6 +4496,15 @@ void Parser::parse_and_visit_if(Parse_Visitor_Base &v) {
 
 parse_maybe_else:
   if (this->peek().type == Token_Type::kw_else) {
+    Token else_token = this->peek();
+    if (previous_in_if_statement &&
```

Comment:
Must fix: This detection of whether we're in a braceless `if` or not is faulty. For example, the following code triggers the diagnostic, but if you remove the outer `if` (first and last lines) there is no diagnostic:
```javascript
if (true) {
   // indentation of 'if' does not match 'else' [E0442]
   if (true) {}
  else console.log()

}
```
Another example of faulty `if` detection:
```javascript
if (true) {
  a.map((o) => {
         if (true) return 1;
    else if (true) return 2;
    else return 3;
  });
}
```